client/lib/plans/index.js

```diff
@@ -143,6 +149,36 @@ export function filterPlansBySiteAndProps( plans, site, hideFreePlan, intervalTy
 	} );
 }
 
+export function getMonthlyPlanByYearly( plan ) {
+	switch ( plan ) {
+		case PLAN_JETPACK_PREMIUM:
+			return PLAN_JETPACK_PREMIUM_MONTHLY;
+		case PLAN_JETPACK_BUSINESS:
+			return PLAN_JETPACK_BUSINESS_MONTHLY;
+		case PLAN_JETPACK_PERSONAL:
+			return PLAN_JETPACK_PERSONAL_MONTHLY;
+		default:
+			return '';
+	}
+}
+
+/**
+ * Returns true if plans match regardeless of their interval.
+ * For example (fake plans):
+ *     isPlanMatch( 'proYearly', 'proYearly' ) => true
+ *     isPlanMatch( 'proYearly', 'proMonthly' ) => true
+ *     isPlanMatch( 'proYearly', 'personalYearly' ) => false
+ *
+ * @param  {String}  planSlugA One of the plan slugs to compare
+ * @param  {String}  planSlugB One of the plan slugs to compare
+ * @return {Boolean}           Whether the plans match
+ */
+export function isPlanMatch( planSlugA, planSlugB ) {
+	return planSlugA === planSlugB ||
```

Comment:
I agree that the following is cleaner and more elegant:

```
return convertToYearly( planSlugA) === convertToYearly( planSlugB);
```

Unfortunately, the way `getYearlyPlanByMonthly` and `getMonthlyPlanByYearly` have been implemented, they return the corresponding version version or `""`. This makes it easy to get false positives and indeed the tests fail when using the suggested implementation. By relying on testing a real slug (not `""`) against a converted slug (which may be `""`), we reduce the chance of these bad results.